program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,62 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from itertools import groupby
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent
+
+
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent CERTAIN intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK
+     * 400: Query returned no results
+    """
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    def get(self, request):
+        """
+        This view should return the best intents per program
+        for the currently authenticated user.
+
+        Best intent is defined as the most recent CERTAIN intent
+        """
+
+        all_intents = list(ProgramIntent.objects.values().order_by('program_uuid'))
+        best_intents = []
+
+        # this is the template for sorting priority for certainty
+        certainty_order = {
+          'CERTAIN_YES' : 1,
+          'CERTAIN_NO' : 1,
+          'MAYBE' : 2
+        }
+
+        # for each program
+        for _program_uuid, intents in groupby(all_intents, lambda x: x['program_uuid']):
```

Comment:
I would extract this into a separate function

then testing can mostly just test the behavior of this function, that it returns the expected intents from an input list - super simple tests to write, no requests or responses

and one test can just test that yes when you do a request it goes through and returns some responses in a happy path case